src/eth_provider/provider.rs

```diff
@@ -826,7 +826,8 @@ where
 
         // This is how reth computes the block size.
         // `https://github.com/paradigmxyz/reth/blob/v0.2.0-beta.5/crates/rpc/rpc-types-compat/src/block.rs#L66`
-        let size = rpc_to_primitive_header(header.clone())?.length();
+        let size =
+            PrimitiveHeader::try_from(header.clone()).ok().ok_or(EthereumDataFormatError::PrimitiveError)?.length();
```

Comment:
Here:
- As `PrimitiveHeader` is only used here please use `reth_primitives::Header` instead and remove it from the import list.
- I think we should use an approach like `PrimitiveHeader::try_from(header.clone()).map_err(|_| EthereumDataFormatError::PrimitiveError)?.length()` (or something like that) because we cannot suppose that the result of the conversion will be fine and we must handle error properly.